homeassistant/components/overkiz/climate_entities/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,203 @@
+"""Support for Somfy Heating Temperature Interface."""
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.climate import (
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+OVERKIZ_TO_PRESET_MODES = {
+    OverkizCommandParam.SECURED: PRESET_AWAY,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.FREE: PRESET_NONE,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+OVERKIZ_TO_HVAC_MODES = {
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+    OverkizCommandParam.MANU: HVACMode.HEAT_COOL,
+}
+
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_HVAC_ACTION = {
+    OverkizCommandParam.COOLING: HVACAction.COOLING,
+    OverkizCommandParam.HEATING: HVACAction.HEATING,
+}
+
+MAP_PRESET_TEMPERATURES = {
+    PRESET_COMFORT: OverkizState.CORE_COMFORT_ROOM_TEMPERATURE,
+    PRESET_ECO: OverkizState.CORE_ECO_ROOM_TEMPERATURE,
+    PRESET_AWAY: OverkizState.CORE_SECURED_POSITION_TEMPERATURE,
+}
+
+MODE_COMMAND_MAPPING = {
+    OverkizCommandParam.COMFORT: OverkizCommand.SET_COMFORT_TEMPERATURE,
+    OverkizCommandParam.ECO: OverkizCommand.SET_ECO_TEMPERATURE,
+    OverkizCommandParam.SECURED: OverkizCommand.SET_SECURED_POSITION_TEMPERATURE,
+}
+
+TEMPERATURE_SENSOR_DEVICE_INDEX = 2
+
+
+class SomfyHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Heating Temperature Interface.
+
+    The thermostat has 3 ways of working:
+      - Auto: Switch to eco/comfort temperature on a schedule (day/hour of the day)
+      - Manual comfort: The thermostat use the temperature of the comfort setting (19°C degree by default)
+      - Manual eco: The thermostat use the temperature of the eco setting (17°C by default)
+      - Freeze protection: The thermostat use the temperature of the freeze protection (7°C by default)
+
+    There's also the possibility to change the working mode, this can be used to change from a heated
+    floor to a cooling floor in the summer.
+    """
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    # Both min and max temp values have been retrieved from the Somfy Application.
+    _attr_min_temp = 15.0
+    _attr_max_temp = 26.0
```

Comment:
And therefore can't be get programmatically ?

Reply:
I assume not, we tried to gather this from attributes where available, but many devices don't have these attributes.

@egguy do you have your diagnostics file? We can have a look.

Reply:
Have checked the diagnostics file, unfortunately these values do not exists programmatically.